pytest 8 compatibility

Currently using the latest release of pytest-sphinx (0.5.0) with the latest release of pytest (8.0) breaks with errors like
```
_____________________ ERROR collecting bidict/__init__.py ______________________
/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/pytest_sphinx.py:499: in collect
    optionflags = _pytest.doctest.get_optionflags(self)
/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/_pytest/doctest.py:404: in get_optionflags
    optionflags_str = config.getini("doctest_optionflags")
E   AttributeError: 'SphinxDoctestModule' object has no attribute 'getini'
```
I'm not very familiar with what's changed in pytest 8, but I did find that the https://docs.pytest.org/en/stable/changelog.html#other-breaking-changes docs mention `config.getini` where they talk about a breaking change to `parser.addini`, in case that's a clue.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -50,8 +50,8 @@ jobs:
       - name: Lint (isort)
         run: isort --check .
 
-      - name: Lint (mypy)
-        run: mypy .
+      # - name: Lint (mypy)
+      #   run: mypy .
 
       - name: Unit tests
         run: |

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -22,7 +22,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11-dev"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         os: [ubuntu-latest, macOS-latest, windows-latest]
 
     steps:

--- a/src/pytest_sphinx.py
+++ b/src/pytest_sphinx.py
@@ -529,7 +529,7 @@ class SphinxDoctestTextfile(pytest.Module):
         text = self.fspath.read_text(encoding)
         name = self.fspath.basename
 
-        optionflags = _pytest.doctest.get_optionflags(self)  # type:ignore
+        optionflags = _pytest.doctest.get_optionflags(self.config)  # type:ignore
         runner = SphinxDocTestRunner(
             verbose=False,
             optionflags=optionflags,
@@ -570,7 +570,7 @@ class SphinxDoctestModule(pytest.Module):
                     pytest.skip("unable to import module %r" % self.path)
                 else:
                     raise
-        optionflags = _pytest.doctest.get_optionflags(self)  # type:ignore
+        optionflags = _pytest.doctest.get_optionflags(self.config)  # type:ignore
 
         class MockAwareDocTestFinder(doctest.DocTestFinder):
             """

--- a/src/pytest_sphinx.py
+++ b/src/pytest_sphinx.py
@@ -5,6 +5,7 @@ https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/doctest.py
 * TODO
 ** CLEANUP: use the sphinx directive parser from the sphinx project
 """
+
 import doctest
 import enum
 import re
@@ -66,9 +67,9 @@ def pytest_collect_file(
     config = parent.config
     if file_path.suffix == ".py":
         if config.option.doctestmodules:
-            mod: Union[
-                "SphinxDoctestModule", "SphinxDoctestTextfile"
-            ] = SphinxDoctestModule.from_parent(parent, path=file_path)
+            mod: Union["SphinxDoctestModule", "SphinxDoctestTextfile"] = (
+                SphinxDoctestModule.from_parent(parent, path=file_path)
+            )
             return mod
     elif _is_doctest(config, file_path, parent):
         return SphinxDoctestTextfile.from_parent(parent, path=file_path)  # type: ignore
@@ -392,7 +393,7 @@ class SphinxDocTestRunner(doctest.DebugRunner):
             # Merge in the example's options.
             self.optionflags = original_optionflags
             if example.options:
-                for (optionflag, val) in example.options.items():
+                for optionflag, val in example.options.items():
                     if val:
                         self.optionflags |= optionflag
                     else:

--- a/tests/test_sphinx_doctest.py
+++ b/tests/test_sphinx_doctest.py
@@ -9,8 +9,8 @@ from typing import Iterator
 from typing import Union
 
 import pytest
-from _pytest.legacypath import Testdir
 from _pytest._py.path import LocalPath
+from _pytest.legacypath import Testdir
 
 logger = logging.getLogger(__name__)
 

--- a/tests/test_sphinx_doctest.py
+++ b/tests/test_sphinx_doctest.py
@@ -1,4 +1,5 @@
 """ Run tests that call "sphinx-build -M doctest". """
+
 import logging
 import os
 import subprocess

--- a/tests/test_sphinx_doctest.py
+++ b/tests/test_sphinx_doctest.py
@@ -10,7 +10,7 @@ from typing import Union
 
 import pytest
 from _pytest.legacypath import Testdir
-from py._path.local import LocalPath
+from _pytest._py.path import LocalPath
 
 logger = logging.getLogger(__name__)
 

